components/collator/src/comparison.rs

```diff
@@ -145,25 +146,61 @@ impl Collator {
             + ?Sized,
     {
         let req = DataRequest {
-            locale,
-            ..Default::default()
+            langid: &locale.id,
+            key_attributes: &locale
+                .keywords
+                .get(&key!("co"))
+                .map(DataKeyAttributes::from_unicode_value)
+                .unwrap_or_default(),
+            metadata: {
+                let mut metadata = DataRequestMetadata::default();
+                metadata.silent = true;
+                metadata
+            },
         };
 
-        let metadata_payload: DataPayload<crate::provider::CollationMetadataV1Marker> =
-            provider.load(req)?.take_payload()?;
+        let metadata_payload: DataPayload<crate::provider::CollationMetadataV1Marker> = provider
+            .load(req)
+            .or_else(|_| {
+                provider.load(DataRequest {
+                    langid: req.langid,
+                    ..Default::default()
+                })
+            })?
+            .take_payload()?;
```

Comment:
Suggestion, here and elsewhere: perhaps refactor to make more clear why you are making two requests

```suggestion
        let metadata_payload: DataPayload<crate::provider::CollationMetadataV1Marker> = provider
            .load(req_specific)
            .or_else(|_| {
                provider.load(req_default)
            })?
            .take_payload()?;
```